### ISSUE TITLE
`Layout/SpaceAroundOperators` - undefined method `source_range` for `nil:NilClass`

### ISSUE THREAD

**Issue by snutij** [rubocop/rubocop#12007]
Hello 👋  I have `Layout/SpaceAroundOperators` failures for the same reason, on several fixtures. As previously, if you prefer an issue for each fixture, feel free to tell me 🙏  

Given a file `foo.rb`:

```ruby
A::*
A::**
A::~
A::&
A::`
A::!=
A::^
A::==
A::===
A::=~
A::>
A::>=
A::>>
A::<<
A::<
A::<=>
A::<<
A::-
A::%
A::|
A::+
A::/


```

Which is a valid syntax: 

```
ruby -c foo.rb
Syntax OK
```

--------

## Expected behavior

Should not trigger an error.

## Actual behavior

`bin/rubocop -d --only Layout/SpaceAroundOperators foo.rb`
 
```
22 errors occurred:
An error occurred while Layout/SpaceAroundOperators cop was inspecting foo.rb:1:0.
An error occurred while Layout/SpaceAroundOperators cop was inspecting foo.rb:2:0.
An error occurred while Layout/SpaceAroundOperators cop was inspecting foo.rb:3:0.
An error occurred while Layout/SpaceAroundOperators cop was inspecting foo.rb:4:0.
An error occurred while Layout/SpaceAroundOperators cop was inspecting foo.rb:5:0.
An error occurred while Layout/SpaceAroundOperators cop was inspecting foo.rb:6:0.
An error occurred while Layout/SpaceAroundOperators cop was inspecting foo.rb:7:0.
An error occurred while Layout/SpaceAroundOperators cop was inspecting foo.rb:8:0.
An error occurred while Layout/SpaceAroundOperators cop was inspecting foo.rb:9:0.
An error occurred while Layout/SpaceAroundOperators cop was inspecting foo.rb:10:0.
An error occurred while Layout/SpaceAroundOperators cop was inspecting foo.rb:11:0.
An error occurred while Layout/SpaceAroundOperators cop was inspecting foo.rb:12:0.
An error occurred while Layout/SpaceAroundOperators cop was inspecting foo.rb:13:0.
An error occurred while Layout/SpaceAroundOperators cop was inspecting foo.rb:14:0.
An error occurred while Layout/SpaceAroundOperators cop was inspecting foo.rb:15:0.
An error occurred while Layout/SpaceAroundOperators cop was inspecting foo.rb:16:0.
An error occurred while Layout/SpaceAroundOperators cop was inspecting foo.rb:17:0.
An error occurred while Layout/SpaceAroundOperators cop was inspecting foo.rb:18:0.
An error occurred while Layout/SpaceAroundOperators cop was inspecting foo.rb:19:0.
An error occurred while Layout/SpaceAroundOperators cop was inspecting foo.rb:20:0.
An error occurred while Layout/SpaceAroundOperators cop was inspecting foo.rb:21:0.
An error occurred while Layout/SpaceAroundOperators cop was inspecting foo.rb:22:0.
```

All stack trace were similar, example for the last error: 

```
An error occurred while Layout/SpaceAroundOperators cop was inspecting foo.rb:22:0.
undefined method `source_range' for nil:NilClass
/Users/snutij/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.53.1/lib/rubocop/cop/layout/space_around_operators.rb:99:in `on_send'
/Users/snutij/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.53.1/lib/rubocop/cop/commissioner.rb:107:in `public_send'
```

## Steps to reproduce the problem

In `RuboCop::Cop::Layout::SpaceAroundOperators` the method `on_send` is trigger, 

Which receive a node: 

```ruby
(rdbg) node
s(:send,
  s(:const, nil, :A), :*)
(ruby) node.source
"A::*"
```

There is no early return because: 

```ruby
(ruby) rational_literal?(node)
nil
```

Then it's enter the elsif condition because:

```ruby
(ruby) node.setter_method?
nil
(ruby) regular_operator?(node)
true
```

Where its trigger the error because `first_argument` is nil

```ruby
(ruby) node.loc.selector
#<Parser::Source::Range foo.rb 3...4>
(ruby) node.first_argument
nil
```

https://github.com/rubocop/rubocop/blob/7e8c47d0e943ee6bce839f7ae5159f1b50b25607/lib/rubocop/cop/layout/space_around_operators.rb#L93-L101

## RuboCop version

```
bundle exec rubocop -V

1.53.1 (using Parser 3.2.2.3, rubocop-ast 1.29.0, running on ruby 3.2.2) [arm64-darwin22]
  - rubocop-minitest 0.31.0
  - rubocop-rake 0.6.0
  - rubocop-sorbet 0.7.0
```
